src/app/public/modules/phone-field/phone-field.component.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  OnDestroy
+} from '@angular/core';
+
+require('intl-tel-input/build/js/utils');
+require('intl-tel-input/build/js/intlTelInput');
+
+/**
+ * NOTE: We can not type these due the the current @types/intl-tel-input version having an
+ * undeclared type which causes linting errors.
+ */
+declare var intlTelInputUtils: any;
+declare var intlTelInputGlobals: any;
+
+import {
+  SkyCountryData
+} from './types';
+
+@Component({
+  selector: 'sky-phone-field',
+  templateUrl: './phone-field.component.html',
+  styleUrls: ['./phone-field.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyPhoneFieldComponent implements OnDestroy {
+
+  public countryData: SkyCountryData[];
+
+  public defaultCountryData: SkyCountryData;
+
+  public disabled: boolean;
+
+  public set selectedCountry(newCountry: SkyCountryData) {
+    if (this._selectedCountry !== newCountry) {
+      this._selectedCountry = newCountry;
+
+      if (!this._selectedCountry.placeholder) {
+        this._selectedCountry.placeholder = intlTelInputUtils.getExampleNumber(
+          newCountry.iso2,
+          true,
+          intlTelInputUtils.numberType.FIXED_LINE
+        );
+      }
+
+      this.sortCountriesWithSelectedAndDefault(newCountry);
+
+      this.selectedCountryChange.emit(newCountry);
+    }
+  }
+
+  public get selectedCountry() {
+    return this._selectedCountry;
```

Comment:
Just checking: no need to return a default value here?

Reply:
Nope. We set the selected country on construction to the first country in the country data and then the directive sets the default country if neeeded.